### PR TITLE
DM creation cleanup

### PR DESCRIPTION
**Pull request by jplatte**
Follow-up to #1665.

### CODE REVIEW

File: crates/matrix-sdk/src/client/mod.rs [hunk 1]
```diff
@@ -1711,6 +1711,15 @@ impl Client {
         Ok(joined_room)
     }
 
+    pub(crate) async fn create_dm_room(&self, user_id: &UserId) -> Result<room::Joined> {
```
Comment (poljar) [inline crates/matrix-sdk/src/client/mod.rs:1714]: Perhaps we should make this public, this way the CI won't complain about dead code. Maybe rename it into `create_dm()`? And of course document it.

**Review by poljar — APPROVED** [review]

**Comment by jplatte** [comment]
@poljar please re-review because of documentation changes

File: crates/matrix-sdk/src/client/mod.rs [hunk 2]
```diff
@@ -1662,17 +1662,13 @@ impl Client {
...
     ///
-    /// Sends a request to `/_matrix/client/r0/createRoom`, returns a
-    /// `create_room::Response`, this is an empty response.
-    ///
-    /// # Arguments
-    ///
-    /// * `room` - The easiest way to create this request is using the
-    /// `create_room::Request` itself.
+    /// Sends a request to `/_matrix/client/r0/createRoom`, returns the created
+    /// room as a [`room::Joined`] object.
```
Comment (poljar) [inline crates/matrix-sdk/src/client/mod.rs:1668]: Perhaps we should link to the `create_dm()` method from here. I know they are one above the other, but that isn't guaranteed to stay this way and people might not have enough screen real estate to see both of them.

**Review by poljar — APPROVED** [review]
Found a rogue `s`, and another nit. looks fine otherwise.

**Pull request merged**